source/dem/dem.cc

```diff
@@ -494,6 +527,13 @@ DEMSolver<dim>::solve()
     boundary_cells_with_lines,
     boundary_cells_with_points);
 
+  // Finding cells adjacent to floating walls
+  boundary_cell_object.find_boundary_cells_for_floating_walls(
+    triangulation,
+    parameters.floating_walls,
+    boundary_cells_for_floating_walls,
+    GridTools::minimal_cell_diameter(triangulation));
+
```

Comment:
Shouldnt you use the largest cell diameter and not the minimal? Otherwise you might miss cells if there is local refinement close to the floating wall. I presume all of your test were with a regular mesh.